zyte_common_items/items.py

```diff
@@ -0,0 +1,254 @@
+    gtin: Optional[List[Gtin]] = None
+
+    #: All product images.
+    #:
+    #: The main image (see ``mainImage``) should be first in the list.
+    #:
+    #: Images only displayed as part of the product description are excluded.
+    images: Optional[List[Image]] = None
+
+    #: Main product image.
+    mainImage: Optional[Image] = None
+
+    #: `Manufacturer part number (MPN)`_.
+    #:
+    #: A product should have the same MPN across different e-commerce websites.
+    #:
+    #: See also: ``gtin``, ``productId``, ``sku``.
+    #:
+    #: .. _Manufacturer part number (MPN): https://en.wikipedia.org/wiki/Part_number
+    mpn: Optional[str] = None
+
+    #: Name as it appears on the product webpage (no post-processing).
+    name: Optional[str] = None
+
+    #: Price at which the product is being offered.
+    #:
+    #: It is a string with the price amount, with a full stop as decimal
+    #: separator, and no thousands separator or currency (see ``currency`` and
+    #: ``currencyRaw``), e.g. ``"10500.99"``.
+    #:
+    #: If ``regularPrice`` is not ``None``, ``price`` should always be lower
+    #: than ``regularPrice``.
+    price: Optional[str] = None
+
+    #: Product identifier, unique within an e-commerce website.
+    #:
+    #: It may come in the form of an SKU or any other identifier, a hash, or
+    #: even a URL.
+    #:
+    #: See also: ``gtin``, ``mpn``, ``sku``.
+    productId: Optional[str] = None
+
+    #: Price at which the product was being offered in the past, and which is
+    #: presented as a reference next to the current price.
+    #:
+    #: It may be labeled as the original price, the list price, or the maximum
+    #: retail price for which the product is sold.
+    #:
+    #: See ``price`` for format details.
+    #:
+    #: If ``regularPrice`` is not ``None``, it should always be higher than
+    #: ``price``.
+    regularPrice: Optional[str] = None
+
+    #: Size or dimensions.
+    #:
+    #: Pertinent to products such as garments, shoes, accessories, etc.
+    #:
+    #: It is extracted as displayed (e.g. ``"XL"``).
+    #:
+    #: See also ``color``, ``style``.
+    size: Optional[str] = None
+
+    #: `Stock keeping unit (SKU)`_ identifier, i.e. a merchant-specific product
+    #: identifier.
+    #:
+    #: See also: ``gtin``, ``mpn``, ``productId``.
+    #:
+    #: .. _Stock keeping unit (SKU): https://en.wikipedia.org/wiki/Stock_keeping_unit
+    sku: Optional[str] = None
+
+    #: Style.
+    #:
+    #: Pertinent to products such as garments, shoes, accessories, etc.
+    #:
+    #: It is extracted as displayed (e.g. ``"polka dots"``).
+    #:
+    #: See also ``color``, ``size``.
+    style: Optional[str] = None
+
+
+@attrs.define(slots=True, kw_only=True)
+class ProductVariant(_ProductBase):
+    """:class:`Product` variant.
+
+    See :attr:`Product.variants`.
+    """
+
+    #: Main URL from which the product variant data was extracted.
```

Comment:
It should be an URL where the complete data about a variant could be extracted, not an URL from which the data in ProductVariant record was extracted

Reply:
How is it different from `canonicalUrl`?